### PR TITLE
Integrate EJB 0.4 updates

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
The individual changes has already been reviewed (see the PR refs in commit messages).

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
@slowli , would you integrate this one, please?

File: src/get-started/java-binding.md [hunk 1]
```diff
@@ -222,20 +241,78 @@ reference. Light clients also provide access to information on the
 [transaction][exonum-transaction] execution result
 (which may be either success or failure) to their users.
 
+An implementation of the `Transaction#execute` method must be a pure function,
+i.e. it must produce the same _observable_ result on all the nodes of the system
+for the given transaction. An observable result is the one that affects the
+blockchain state hash:<!-- TODO: link the definition? -->
```
Comment (slowli) [inline src/get-started/java-binding.md]: Should this TODO item be removed, or can you resolve it? The glossary does contain a definition for the blockchain state.

File: src/get-started/java-binding.md [hunk 2]
```diff
@@ -479,27 +556,31 @@ For using the library just include the dependency in your `pom.xml`:
...
+[service-after-commit]: https://exonum.com/doc/api/java-binding-core/0.4/com/exonum/binding/service/Service.html#afterCommit(com.exonum.binding.service.BlockCommittedEvent)
+[node-submit-transaction]: https://exonum.com/doc/api/java-binding-core/0.4/com/exonum/binding/service/Node.html#submitTransaction(com.exonum.binding.transaction.RawTransaction)
+[standardserializers]: https://exonum.com/doc/api/java-binding-common/0.4/com/exonum/binding/common/serialization/StandardSerializers.html
+[storage-indices]: https://exonum.com/doc/api/java-binding-core/0.4/com/exonum/binding/storage/indices/package-summary.html
+[transaction]: https://exonum.com/doc/api/java-binding-core/0.4/com/exonum/binding/transaction/Transaction.html
+[transaction-execution-context]: https://exonum.com/doc/api/java-binding-core/0.4/com/exonum/binding/transaction/TransactionContext.html
 [transactions]: ../architecture/transactions.md
-[transactionconvererter]: https://exonum.com/doc/api/java-binding-core/latest/com/exonum/binding/service/TransactionConverter.html
+[transactions-messages]: ../architecture/transactions/#messages
```
Comment (slowli) [inline src/get-started/java-binding.md]: Nit: It is better to link to the Markdown document (`../architecture/transactions.md#messages`); the link will be translated during doc compilation.

File: src/get-started/java-binding.md (hunk 1 above)
Comment (dmitry-timofeev) [reply src/get-started/java-binding.md]: I remember that I was dissatisfied with the glossary entry (it didn't clarify things much), hence left a todo.

**Comment by dmitry-timofeev** [comment]
> Minor question: does incompatibility with the light client mentioned in the "Known Limitations" section still hold, given that Exonum core and LC implement Protobuf serialization?

Thanks, they are now compatible (as txs are submitted to the core, not a custom endpoint)!

**Pull request merged**